### ISSUE TITLE
OpenLayers 5.2.0 - setMap fails because Observable.unByKey has been removed

### ISSUE THREAD

**Issue by FBNitro** [walkermatt/ol-layerswitcher#95]
I'm attempting to remove the layerswitcher at runtime, and then reload it later.  I'm doing this because the layerswitcher doesn't seem to pick up new layer groups if I simply re-render (I can create another issue for that separate if you like).

When I attempt to remove it, it fails in setMap(map).  I believe this is because Observable.unByKey has been removed in the later versions.  It is no longer documented here: https://openlayers.org/en/latest/apidoc/module-ol_Observable-Observable.html

The exact error is: "Uncaught TypeError: Observable.unByKey is not a function"

LayerSwitcher code:
``` javascript
    createClass(LayerSwitcher, [{
        key: 'setMap',
        value: function setMap(map) {
            // Clean up listeners associated with the previous map
            for (var i = 0, key; i < this.mapListeners.length; i++) {
                Observable.unByKey(this.mapListeners[i]);
            }
```

**Comment by umbe1987** [comment]
You mean this one? https://openlayers.org/en/latest/apidoc/module-ol_Observable.html#.unByKey

**Comment by FBNitro** [comment]
Yeah that one... sorry I missed it.  Let me try to get a sample together with the error I'm seeing.
